src/v/storage/parser.cc

```diff
@@ -84,30 +84,28 @@ static ss::future<result<iobuf>> verify_read_iobuf(
   size_t expected,
   ss::sstring msg,
   bool recover = false) {
-    return read_iobuf_exactly(in, expected)
-      .then([msg = std::move(msg), expected, recover](iobuf b) {
-          if (likely(b.size_bytes() == expected)) {
-              return ss::make_ready_future<result<iobuf>>(std::move(b));
-          }
-          if (!recover) {
-              stlog.error(
-                "cannot continue parsing. recived size:{} bytes, expected:{} "
-                "bytes. context:{}",
-                b.size_bytes(),
-                expected,
-                msg);
-          } else {
-              stlog.debug(
-                "recovery ended with short read. recived size:{} bytes, "
-                "expected:{} "
-                "bytes. context:{}",
-                b.size_bytes(),
-                expected,
-                msg);
-          }
-          return ss::make_ready_future<result<iobuf>>(
-            parser_errc::input_stream_not_enough_bytes);
-      });
+    auto b = co_await read_iobuf_exactly(in, expected);
+
+    if (likely(b.size_bytes() == expected)) {
+        co_return b;
```

Comment:
co_return std::move(b) makes sense here; I remember copy elision does not apply in the context of coroutines

Reply:
i think in this case it does as the `iobuf` has removed copy constructor

Reply:
You are right, I also didn't remember correctly. it's co_yield that will require the use of std::move